app/src/main/java/org/simple/clinic/widgets/TheActivityLifecycle.kt

```diff
@@ -0,0 +1,9 @@
+package org.simple.clinic.widgets
+
+sealed class TheActivityLifecycle : UiEvent {
```

Comment:
I considered it too, but I wasn't sure if it'll be useful much. We neither have fragments nor multiple activities. There's only one class for which we need arch components for. Are there other advantages?

I haven't used lifecycle components yet.

Reply:
Right. My reasoning is that there are many libraries like Uber's `AutoDispose` which have out of the box support for it, and it's better to have something official which supports what we need instead of rolling out our own, IMO.

Anyway, it's something we can take a look at later.

Reply:
🙂👍